src/modules/twap/containers/TwapFormWidget/index.tsx

```diff
@@ -9,19 +9,25 @@ import { partsStateAtom } from '../../state/partsStateAtom'
 import { DeadlineSelector } from '../../pure/DeadlineSelector'
 import { useMemo } from 'react'
 import { displayTime } from 'utils/displayTime'
-import { orderDeadlines } from '../../const'
+import { DEFAULT_TWAP_SLIPPAGE, orderDeadlines } from '../../const'
+import { twapTimeIntervalAtom } from '../../state/twapOrderAtom'
+import { PrimaryActionButton } from '../../pure/PrimaryActionButton'
+import { useTwapFormActions } from '../../hooks/useTwapFormActions'
+import { useTwapFormState } from '../../hooks/useTwapFormState'
 
 export function TwapFormWidget() {
   const { numberOfPartsValue, slippageValue, deadline, customDeadline, isCustomDeadline } =
     useAtomValue(twapOrdersSettingsAtom)
   const partsState = useAtomValue(partsStateAtom)
+  const timeInterval = useAtomValue(twapTimeIntervalAtom)
   const updateState = useUpdateAtom(updateTwapOrdersSettingsAtom)
 
-  const partsTime = useMemo(() => {
-    const timestamp = isCustomDeadline ? (customDeadline.hours * 60 + customDeadline.minutes) * 60 * 1000 : deadline
+  const formActions = useTwapFormActions()
+  const formState = useTwapFormState()
 
-    return displayTime(timestamp / numberOfPartsValue)
-  }, [numberOfPartsValue, deadline, customDeadline, isCustomDeadline])
+  const partsTime = useMemo(() => {
+    return displayTime((timeInterval * 1000) / numberOfPartsValue)
+  }, [numberOfPartsValue, timeInterval])
```

Comment:
Nitpick (just because I like shorter fns), feel free to ignore:

```suggestion
  const partsTime = useMemo(() => displayTime((timeInterval * 1000) / numberOfPartsValue), [numberOfPartsValue, timeInterval])
```